### PR TITLE
Indexing into TransformChain by name

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
I can't push directly to your branch, here's the code we worked on today

### CODE REVIEW

File: napari/layers/transforms.py [hunk 1]
```diff
@@ -41,14 +41,22 @@ def inverse(self):
...
+            basetype=Transform,
+            iterable=transforms,
+            lookup={str: lambda q, e: q == e.name},
+        )
 
     def __call__(self, coords):
         return tz.pipe(coords, *self)
 
+    def __newlike__(self, iterable):
+        return ListModel(self._basetype, iterable, self._lookup)
```
Comment (jni) [inline napari/layers/transforms.py:56]: @kne42 would you mind clarifying what these lines do? We copied them from LayerList but we don't really understand what's happening. (At least I don't!)

File: napari/layers/transforms.py [hunk 2]
```diff
@@ -41,14 +41,22 @@ def inverse(self):
...
+
     def set_slice(self, axes: Sequence[int]) -> TransformChain:
+        self._set_transform_attributes()
```
Comment (jni) [inline napari/layers/transforms.py]: @GenevieveBuckley this line needs to be removed!

Comment (jni) [reply napari/layers/transforms.py]:
```suggestion
```

**Review by jni — APPROVED** [review]

File: napari/layers/transforms.py (hunk 1 above)
Comment (kne42) [reply napari/layers/transforms.py:56]: __newlike__ is used when you want to create a new List with the same settings but out of a different iterable. it is used in slicing and copy

File: napari/layers/transforms.py (hunk 2 above)
Comment (GenevieveBuckley) [reply napari/layers/transforms.py]: Good catch, thank you!

File: napari/layers/transforms.py (hunk 1 above)
Comment (jni) [reply napari/layers/transforms.py:56]: Is it standard Python (I couldn't find any docs?) or your own invention? If the latter, the common practice is to use single-underscore wrapping, leaving dunder to standard Python protocols.

**Comment by kne42** [comment]
my own invention. can refactor in a future PR

**Pull request merged**